recipes/bitserializer/all/conanfile.py

```diff
@@ -81,14 +103,41 @@ def source(self):
         get(self, **self.conan_data["sources"][self.version],
             destination=self.source_folder, strip_root=True)
 
+    def generate(self):
+        if not self._is_header_only:
+            tc = CMakeToolchain(self)
+            tc.variables["BUILD_CPPRESTJSON_ARCHIVE"] = self.options.with_cpprestsdk
+            tc.variables["BUILD_RAPIDJSON_ARCHIVE"] = self.options.with_rapidjson
+            tc.variables["BUILD_PUGIXML_ARCHIVE"] = self.options.with_pugixml
+            tc.variables["BUILD_RAPIDYAML_ARCHIVE"] = self.options.with_rapidyaml
+            tc.variables["BUILD_CSV_ARCHIVE"] = self.options.with_csv
+            tc.generate()
+            deps = CMakeDeps(self)
+            deps.generate()
+
     def build(self):
-        pass
+        if not self._is_header_only:
+            self._patch_sources()
+            cmake = CMake(self)
+            cmake.configure()
+            cmake.build()
 
     def package(self):
+        if not self._is_header_only:
+            cmake = CMake(self)
+            cmake.install()
+            rmdir(self, os.path.join(self.package_folder, "share"))
+        else:
+            copy(self, "*.h", src=os.path.join(self.source_folder, "include"), dst=os.path.join(self.package_folder, "include"))
+        # Copy license
         copy(self, "license.txt", src=self.source_folder, dst=os.path.join(self.package_folder, "licenses"))
-        copy(self, "*.h", src=os.path.join(self.source_folder, "include"), dst=os.path.join(self.package_folder, "include"))
+
+    def package_id(self):
+        if self._is_header_only:
+            self.info.header_only()
 
     def package_info(self):
+        lib_suffix = "d" if self.info.settings.build_type == "Debug" else ""
```

Comment:
Where does you use it? I can't find 🤔

Reply:
Line 184:
`            self.cpp_info.components["bitserializer-csv"].libs = [ f"csv-archive{lib_suffix}" ]`

Reply:
Now it makes sense, thank you!